Microsoft.Toolkit.Uwp.UI.Controls/BladeView/BladeView.Properties.cs

```diff
@@ -27,12 +29,66 @@ public partial class BladeView
         public static readonly DependencyProperty ActiveBladesProperty = DependencyProperty.Register(nameof(ActiveBlades), typeof(IList<BladeItem>), typeof(BladeView), new PropertyMetadata(null));
 
         /// <summary>
+        /// Identifies the <see cref="FullScreenBlades"/> attached property.
+        /// </summary>
+        public static readonly DependencyProperty FullScreenBladesProperty = DependencyProperty.RegisterAttached(nameof(FullScreenBlades), typeof(bool), typeof(BladeView), new PropertyMetadata(false, FullScreenBladesChangedCallback));
+
+        /// <summary>
         /// Gets or sets a collection of visible blades
         /// </summary>
         public IList<BladeItem> ActiveBlades
         {
             get { return (IList<BladeItem>)GetValue(ActiveBladesProperty); }
             set { SetValue(ActiveBladesProperty, value); }
         }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether blades are full screen
+        /// </summary>
+        public bool FullScreenBlades
+        {
+            get { return (bool)GetValue(FullScreenBladesProperty); }
+            set { SetValue(FullScreenBladesProperty, value); }
+        }
+
+        private static void FullScreenBladesChangedCallback(DependencyObject dependencyObject, DependencyPropertyChangedEventArgs e)
+        {
+            // Check if value changed
+            if (e.NewValue != e.OldValue)
+            {
+                var bladeView = (BladeView)dependencyObject;
+                var bladeScrollViewer = bladeView.GetScrollViewer();
+
+                if (bladeView.FullScreenBlades)
+                {
+                    // Cache previous values of blade items properties (width & height)
+                    bladeView._cachedBladeItemSizes.Clear();
+                    foreach (BladeItem bladeItem in bladeView.Items)
+                    {
+                        bladeView._cachedBladeItemSizes.Add(bladeItem, new Vector2((float)bladeItem.Width, (float)bladeItem.Height));
```

Comment:
Perhaps have the value be `Size` rather than `Vector2`. Size provides `Width` and `Height` properties